subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ArtifactTransformsReportTask.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics;
+
+import groovy.lang.Closure;
+import groovy.lang.DelegatesTo;
+import org.gradle.api.Action;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.Incubating;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.transform.TransformAction;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Property;
+import org.gradle.api.reporting.Reporting;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.diagnostics.artifacttransforms.ArtifactTransformReports;
+import org.gradle.api.tasks.diagnostics.internal.artifacttransforms.ArtifactTransformReportsImpl;
+import org.gradle.api.tasks.diagnostics.internal.artifacttransforms.model.ArtifactTransformReportModel;
+import org.gradle.api.tasks.diagnostics.internal.artifacttransforms.model.ArtifactTransformReportModelFactory;
+import org.gradle.api.tasks.diagnostics.internal.artifacttransforms.renderer.ConsoleArtifactTransformReportRenderer;
+import org.gradle.api.tasks.diagnostics.internal.artifacttransforms.spec.ArtifactTransformReportSpec;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.internal.logging.text.StyledTextOutput;
+import org.gradle.internal.logging.text.StyledTextOutputFactory;
+import org.gradle.internal.serialization.Cached;
+import org.gradle.util.internal.ClosureBackedAction;
+import org.gradle.work.DisableCachingByDefault;
+
+import javax.inject.Inject;
+
+/**
+ * A task which reports information about the Artifact Transforms (implemented by {@link TransformAction}) used by a project.
+ *
+ * This is useful for investigating ambiguous transformation scenarios.  The output can help predict which ATs will need
+ * to be modified to remove ambiguity.  This class implements {@link Reporting} to make configuring additional file output formats simple.
+ *
+ * @since 8.12
+ */
+@Incubating
+@DisableCachingByDefault(because = "Produces only non-cacheable console output")
+public abstract class ArtifactTransformsReportTask extends DefaultTask implements Reporting<ArtifactTransformReports> {
```

Comment:
There are a couple weird things in here that seems to stem from implementing `Reporting`. 

They include:
- Needing to implement a `Closure`-accepting method
- The `Action`-accepting method returning the reports object being configured

But most notably:
- The reports object, `ArtifactTransformReports`, has no reports.

Why implement `Reports` if we provide no reports? Is there some technical reason we need to implement that interface? If so, are there any straightforward changes we can make to that requirement which would make it so we don't need to implement `Reporting`? This interface is from 2012 and has some odd patterns which would be nice to avoid